### PR TITLE
Prevent Updates to Unchanged Location Records

### PR DESCRIPTION
**Pull request by bencap**
This change compares incoming values with existing values in the `warehouse.location` table
to decide if there is a need to perform an update. Unnecessary updates add bloat to the
table, which this change will cut down on.

### CODE REVIEW

**Comment by bencap** [comment]
[Investigate non-sequential PK increments in warehouse tables (encounters, individuals)](https://trello.com/c/p7WNVOVI/1336-investigate-non-sequential-pk-increments-in-warehouse-tables-encounters-individuals)

File: lib/id3c/cli/command/etl/__init__.py [hunk 1]
```diff
@@ -357,19 +357,70 @@ def upsert_location(db: DatabaseSession,
...
+    }
+
+    with db.cursor() as cursor:
+        cursor.execute("""
+            select
+                location_id as id,
+                scale,
+                identifier,
+                hierarchy,
+                (location.hierarchy) != (%(hierarchy)s) as data_changed
```
Comment (davereinhart) [inline lib/id3c/cli/command/etl/__init__.py:373]: We'll want to cast the incoming value to `hstore` data type for the comparison. The extra parenthesis aren't necessary.

**Review by davereinhart — CHANGES_REQUESTED** [review]

**Review by davereinhart — COMMENTED** [review]
~~@bencap I noticed some interesting behavior when testing with a FHIR bundle containing multiple encounters. An update on the location record (when data has changed) happens for every encounter. I suspect that's because the changes aren't committed until the entire bundle is processed, but I haven't confirmed that yet.~~ Disregard, appears be to specific to the process I was using to test it, subsequent tests look good.

**Review by davereinhart — APPROVED** [review]

**Comment by bencap** [comment]
Deployed.

**Pull request merged**